splunklib/searchcommands/search_command.py

```diff
@@ -852,15 +852,23 @@ def _execute(self, ifile, process):
     @staticmethod
     def _read_chunk(ifile):
         # noinspection PyBroadException
+        if six.PY2:
+            istream = ifile
+        else:
+            try:
+                istream = ifile.buffer
```

Comment:
I think this logic could be moved to `_process_protocol_v2` before calling `_read_chunk`, since it doesn't need to be re-evaluated for each chunk.

Also, since intermixing reads from `sys.stdin` and `sys.stdin.buffer` does not work well (presumably due to buffering, but I haven't checked), it might be safer to use`ifile = ifile.buffer` so that the wrong variable can't be used by mistake. (It also means the assignment for python2 can be skipped entirely.)

Reply:
Moved this logic into `_as_binary_stream` and created a Python 3-only assert to ensure that incoming objects come from the binary stream hierarchy.